tests/test_metrics_api.py

```diff
@@ -0,0 +1,362 @@
+        assert aggregation["total_seconds"] == self.seconds
+
+    def test_retrieval_counter_not_present(self) -> None:
+        ABSENT_METRIC_ID = 4096
+        query_str = self.build_simple_query(metric_id=ABSENT_METRIC_ID)
+        response = self.app.post(
+            SNQL_ROUTE, data=json.dumps({"query": query_str, "dataset": "metrics"})
+        )
+        data = json.loads(response.data)
+
+        assert response.status_code == 200
+        assert len(data["data"]) == 0, data
+
+
+class TestMetricsApiSets(BaseApiTest):
+    @pytest.fixture
+    def test_app(self) -> Any:
+        return self.app
+
+    @pytest.fixture
+    def test_entity(self) -> Union[str, Tuple[str, str]]:
+        return "metrics_sets"
+
+    @pytest.fixture(autouse=True)
+    def setup_post(self, _build_snql_post_methods: Callable[[str], Any]) -> None:
+        self.post = _build_snql_post_methods
+
+    def setup_method(self, test_method: Any) -> None:
+        super().setup_method(test_method)
+
+        # values for test data
+        self.metric_id = 1002
+        self.org_id = 103
+        self.project_ids = [1, 2]  # 2 projects
+        self.seconds = 180 * 60
+
+        self.default_tags = {
+            TAG_1_KEY: TAG_1_VALUE_1,
+            TAG_2_KEY: TAG_2_VALUE_1,
+            TAG_3_KEY: TAG_3_VALUE_1,
+            TAG_4_KEY: TAG_4_VALUE_1,
+        }
+        self.skew = timedelta(seconds=self.seconds)
+
+        self.base_time = datetime.utcnow().replace(
+            minute=0, second=0, microsecond=0, tzinfo=pytz.utc
+        ) - timedelta(minutes=self.seconds)
+        self.storage = get_writable_storage(StorageKey.METRICS_BUCKETS)
+        self.unique_set_values = 100
+        self.generate_sets()
+
+    def teardown_method(self, test_method: Any) -> None:
+        teardown_common()
+
+    def generate_sets(self) -> None:
+        events = []
+        processor = self.storage.get_table_writer().get_stream_loader().get_processor()
+
+        for n in range(self.seconds):
+            for p in self.project_ids:
+                msg = {
+                    "org_id": self.org_id,
+                    "project_id": p,
+                    "type": METRICS_SET_TYPE,
+                    "value": [n % self.unique_set_values],
+                    "timestamp": self.base_time.timestamp() + n,
+                    "tags": self.default_tags,
+                    "metric_id": self.metric_id,
+                    "retention_days": RETENTION_DAYS,
+                }
+
+                processed = processor.process_message(
+                    msg, KafkaMessageMetadata(0, 0, self.base_time),
+                )
+                if processed:
+                    events.append(processed)
+        write_processed_messages(self.storage, events)
+
+    def test_sets_basic(self) -> None:
+        query_str = """MATCH (metrics_sets)
+                    SELECT uniqCombined64Merge(value) AS unique_values BY project_id, org_id
```

Comment:
same question here about using `uniqCombined64Merge` vs just `uniq` given the translation mappers

Reply:
same answer as above, I'll make that change